community/kernel/src/main/java/org/neo4j/kernel/configuration/Config.java

```diff
@@ -303,6 +317,79 @@ else if ( unit.equalsIgnoreCase( "g" ) )
     }
 
     /**
+     * This mechanism can be used as an argument to {@link #view(Function)} to view a set of config options that share a common base config key as a group.
+     * This specific version handles multiple groups, so the common base key should be followed by a number denoting the group, followed by the group config
+     * values, eg:
+     *
+     * {@code <base name>.<group key>.<config key>}
+     *
+     * The config of each group can then be accessed as if the {@code config key} in the pattern above was the entire config key. For example, given the
+     * following configuration:
+     *
+     * <pre>
+     *     dbms.books.0.name=Hansel & Gretel
+     *     dbms.books.0.author=JJ Abrams
+     *     dbms.books.1.name=NKJV
+     *     dbms.books.1.author=Jesus
+     * </pre>
+     *
+     * We can then access these config values as groups:
+     *
+     * <pre>
+     * {@code
+     *     Setting<String> bookName = setting("name", STRING); // note that the key here is only 'name'
+     *
+     *     ConfigView firstBook = config.view( groups("dbms.books") ).get(0);
+     *
+     *     assert firstBook.get(bookName).equals("Hansel & Gretel");
+     * }
+     * </pre>
+     *
+     * @param baseName the base name for the groups, this will be the first part of the config key, followed by a grouping number, followed by the group
+     *                 config options
+     * @return a list of grouped config options
+     */
+    public static Function<ConfigValues,List<ConfigView>> groups( String baseName )
```

Comment:
Please return a `java.util.function.Function` instead. The `org.neo4j.function.*` classes and interfaces are internal, and in the process of being removed.